lightning/src/ln/channelmanager.rs

```diff
@@ -413,6 +414,23 @@ pub struct PaymentId(pub [u8; Self::LENGTH]);
 impl PaymentId {
 	/// Number of bytes in the id.
 	pub const LENGTH: usize = 32;
+
+
+	/// Constructs an HMAC to include in [`OffersContext::OutboundPayment`] for the payment id
+	/// along with the given [`Nonce`].
+	pub fn hmac_for_offer(
```

Comment:
re: `*_for_offer` -- this may also be used for refunds, right? May be easier to name it in relation to invoice errors since that's what it's used for?

Reply:
We don't set the HMAC in refunds because (1) it would increase the QR code size and (2) we don't necessarily want someone to send an `InvoiceError` to one in response -- rather the `Refund` creator can set a short expiry or abandon it manually.

Reply:
Re: naming -- how about `hmac_using_nonce`?

Reply:
I think I do prefer that name... `hmac_for_offer_payment` also works for me.

Reply:
`hmac_for_offer_payment` SGTM